scalafix-core/shared/src/main/scala/scalafix/internal/config/Config.scala

```diff
@@ -0,0 +1,22 @@
+package scalafix.internal.config
+
+import metaconfig.ConfDecoder
+import org.langmeta._
+
+trait Config[A] {
```

Comment:
This does solve my complaints in https://github.com/scalacenter/scalafix/pull/355#discussion_r139296486, so we can give it a try 😄

Reply:
I also share your concern about generalizing to early, I wanted to try this way because with NoInfer and Disable was literally copy/paste. If you know already some example that might have critical custom fields I could give it a try and see how it works, before committing to this.

Reply:
see https://github.com/scalacenter/scalafix/pull/438